3-enrich/beam-enrich/README.md

```diff
@@ -0,0 +1,148 @@
+enrich them and store them into an output PubSub topic.
+Events are enriched using the [scala-common-enrich][common-enrich] library.
+
+## Building
+
+This project uses [sbt-native-packager][sbt-native-packager].
+
+### Zip archive
+
+To build the zip archive, run:
+
+```bash
+sbt universal:packageBin
+```
+
+### Docker image
+
+To build a Docker image, run:
+
+```bash
+sbt docker:publishLocal
+```
+
+## Running
+
+### Directly
+
+Once unzipped the artifact can be run as follows:
+
+```bash
+./bin/beam-enrich \
+  --runner=DataFlowRunner \
+  --project=project-id \
+  --streaming=true \
+  --zone=europe-west2-a \
+  --gcpTempLocation=gs://location/ \
+  --raw=projects/project/subscriptions/raw-topic-subscription \
+  --enriched=projects/project/topics/enriched-topic \
+  --bad=projects/project/topics/bad-topic \
+  --pii=projects/project/topics/pii-topic \ #OPTIONAL
+  --resolver=iglu_resolver.json \
+  --enrichments=enrichments/
+```
+
+To display the help message:
+
+```bash
+./bin/beam-enrich \
+  --runner=DataFlowRunner \
+  --help
+```
+
+### Through a docker container
+
+A container can be run as follows:
+
+```bash
+docker run \
+  -v $PWD/config:/snowplow/config \
+  -e GOOGLE_APPLICATION_CREDENTIALS=/snowplow/config/credentials.json \ # if running outside GCP
+  snowplow-docker-registry.bintray.io/snowplow/beam-enrich:0.1.0 \
+  --runner=DataFlowRunner \
+  --job-name=snowplow-enrich \
+  --project=project-id \
+  --streaming=true \
+  --zone=europe-west2-a \
+  --gcpTempLocation=gs://location/ \
+  --raw=projects/project/subscriptions/raw-topic-subscription \
+  --enriched=projects/project/topics/enriched-topic \
+  --bad=projects/project/topics/bad-topic \
+  --pii=projects/project/topics/pii-topic \ #OPTIONAL
+  --resolver=/snowplow/config/iglu_resolver.json \
+  --enrichments=/snowplow/config/enrichments/
+```
+
+To display the help message:
+
+```bash
+docker run snowplow-docker-registry.bintray.io/snowplow/beam-enrich:0.1.0 \
+  --runner=DataFlowRunner \
+  --help
+```
+
+### Additional information
+
+Note that, for the enrichments relying on local files, the files need to be accessible from Google
+Cloud Storage, e.g. for the IP lookups enrichment:
+
+```json
+{
+  "schema": "iglu:com.snowplowanalytics.snowplow/ip_lookups/jsonschema/2-0-0",
+  "data": {
+    "name": "ip_lookups",
+    "vendor": "com.snowplowanalytics.snowplow",
+    "enabled": true,
+    "parameters": {
+      "geo": {
+        "database": "GeoLite2-City.mmdb",
+        "uri": "gs://ben-sp-test/maxmind"
+      }
+    }
+  }
+}
+```
+
+A full list of all the Beam CLI options can be found at:
+https://cloud.google.com/dataflow/pipelines/specifying-exec-params#setting-other-cloud-pipeline-options.
+
+## Testing
+
+To run the tests:
+
+```
+sbt test
+```
+
+## REPL
+
+To experiment with the current codebase in [Scio REPL](https://github.com/spotify/scio/wiki/Scio-REPL)
+simply run:
+
+```
+sbt repl/run
```

Comment:
Does scio repl really give us something?

Also don't we need an usual [Find out more](https://github.com/snowplow/snowplow/blob/5e1db1a4d10e9d4174d6c08f897b704d7310e5f1/3-enrich/spark-enrich/README.md#find-out-more) section?

Reply:
I think it's pretty cool to have a repl to try out SCollection operations